parser-typechecker/src/Unison/Codebase/Editor/Git.hs

```diff
@@ -83,26 +84,28 @@ pullBranch repo@(ReadGitRepo uri) = do
     unless isGitDir . throwError $ GitError.UnrecognizableCheckoutDir repo localPath
 
   -- | Do a `git pull` on a cached repo.
-  checkoutExisting :: (MonadIO m, MonadCatch m, MonadError GitProtocolError m)
+  checkoutExisting :: (MonadIO m)
                    => FilePath
                    -> Maybe Text
-                   -> m ()
+                   -> ExceptT GitProtocolError m ()
   checkoutExisting localPath maybeRemoteRef =
     ifM (isEmptyGitRepo localPath)
       -- I don't know how to properly update from an empty remote repo.
       -- As a heuristic, if this cached copy is empty, then the remote might
       -- be too, so this impl. just wipes the cached copy and starts from scratch.
       goFromScratch
       -- Otherwise proceed!
-      (catchIO
-        (withStatus ("Updating cached copy of " ++ Text.unpack uri ++ " ...") $ do
-          -- Fetch only the latest commit, we don't need history.
-          gitIn localPath (["fetch", "origin", remoteRef, "--quiet"] ++ ["--depth", "1"])
-          -- Reset our branch to point at the latest code from the remote.
-          gitIn localPath ["reset", "--hard", "--quiet", "FETCH_HEAD"]
-          -- Wipe out any unwanted files which might be sitting around, but aren't in the commit.
-          gitIn localPath ["clean", "-d", "--force", "--quiet"])
-        (const $ goFromScratch))
+      do
+        succeeded <- liftIO . handleIO (const $ pure False) $ do
```

Comment:
By slightly re-arranging this block we can remove the MonadCatch constraint which propagates through all of SqliteCodebase.